From ef928654ec59ad6a805b588dd5cf3bcb1b3cfa8a Mon Sep 17 00:00:00 2001
From: David Chang <david.chang@hpe.com>
Date: Thu, 16 Jul 2020 16:47:43 +0800
Subject: [PATCH] CVE-2020-7206: Fix php code injection problem

Add escapeshellcmd() to check the embed strings, for resolving
php code injection problem in nagios-plugins-hpilo.
---
 action-url/nagios_hpeilo_service_details.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/action-url/nagios_hpeilo_service_details.php b/action-url/nagios_hpeilo_service_details.php
index 01fd967..31d4653 100644
--- a/action-url/nagios_hpeilo_service_details.php
+++ b/action-url/nagios_hpeilo_service_details.php
@@ -22,7 +22,7 @@
         $ip=$_GET["ip"];
         $comm=$_GET["comm"];
         $id=$_GET["id"];
-        $cmd="./nagios_hpeilo_engine -H $ip -C $comm -o $id -J";
+        $cmd=escapeshellcmd("./nagios_hpeilo_engine -H $ip -C $comm -o $id -J");
         $ret = exec($cmd,$output);
         echo join("\n",$output);
 ?>
